frontend/src/components/IntegrationConfiguration.tsx

```diff
@@ -218,49 +228,59 @@ const SelectBoard: FC<{ configuration: IntegrationConfiguration }> = ({
         <LoadingSpinner />
       ) : (
         <>
-          {(tokenError || errorMessage) && (
+          {(tokenError || boardsTokenError || errorMessage) && (
             <Alert
               severity={errorMessage ? 'error' : 'info'}
               onClose={() => {
-                if (tokenError) setTokenError(false);
+                if (boardsTokenError) setBoardsTokenError(false);
+                if (tokenError) {
+                  boards.refetch();
+                  selected.refetch();
+                  setTokenError(false);
+                }
                 if (errorMessage) setErrorMessage('');
               }}
               icon={false}
             >
               {errorMessage || t('Oauth can not select board')}
             </Alert>
           )}
-          <Select
-            name="board"
-            id="board"
-            className="react-select"
-            classNamePrefix="react-select"
-            styles={{ menuPortal: (base) => ({ ...base, zIndex: 9999 }) }}
-            placeholder="Select board"
-            isDisabled={boards.isError}
-            menuPortalTarget={document.body}
-            onChange={(entry) => {
-              if (entry) {
-                patch({ id, name, roadmapId, boardId: entry.value })
-                  .unwrap()
-                  .catch((err) => {
-                    setErrorMessage(
-                      err.data?.message ?? 'something went wrong',
-                    );
-                  });
-              }
-            }}
-            defaultValue={
-              selected.data && {
-                value: selected.data.id,
-                label: selected.data.name,
+          {!selected.isError && (
```

Comment:
```suggestion
          {selected.isSuccess && (
```

I suppose `isSuccess` could be a better condition, as it can't really be error before the first load, but it doesn't have any data then either.

Same could work better for the skip condition, as that should sequence the call only after other successful call

```ts
  const boards = apiV2.useGetIntegrationBoardsQuery(
    { name, roadmapId },
    { skip: !selected.isSuccess },
  );
```